### ISSUE TITLE
Can the ‘ssd_resnet_50_fpn_coco‘ be converted to a tflite model and deploy in android?

### ISSUE THREAD

**Issue by yzmean** [tensorflow/models#7870]
### System information
- **What is the top-level directory of the model you are using**: object_detection
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: no
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: convert model in Ubuntu 16.04,deploy model in Windows10
- **TensorFlow installed from (source or binary)**: binary, installed from pip: pip install tensorflow-gpu
- **TensorFlow version (use command below)**: 1.14
- **Bazel version (if compiling from source)**: None
- **CUDA/cuDNN version**: CUDA10
- **GPU model and memory**: TITAN XP 12G
- **Phone**: HUAWEI Mate 30 Pro ,Android10 , 8G RAM, HUAWEI Kirin 990
- **Exact command to reproduce**:

### Describe the problem
I use the instruction of https://github.com/tensorflow/models/issues/6709#issuecomment-543432460 to convert ```ssd_resnet_50_fpn_coco``` to ```*.tflite```file. I finally got the file successfully. Here is my [tflite file](https://drive.google.com/open?id=12bp6mxasB9EGo3VPu3L5ljZlGm_rhaeF). But When I deploy the ```.tflite``` file in Android Studio ,the app crashed.

```ssd_resnet_50_fpn_coco``` is download from [Tensorflow detection model zoo](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/detection_model_zoo.md).

Android app code is download from [TensorFlow Lite Object Detection Android Demo](https://github.com/tensorflow/examples/tree/master/lite/examples/object_detection/android).It can install and run in my phone.

Then ready to deploy ```ssd_resnet_50_fpn_coco```tflite model in android. 
- First: I just rename my```.tflite```file to ```detect1.tflite``` and copy to ```app\src\main\assets``` and not change the ```labelmap.txt``` file. Then I modified the ```TF_OD_API_INPUT_SIZE``` parameter from 300 to 640 in ```\app\src\main\java\org\tensorflow\lite\examples\detection\DetectorActivity.java``` .

However, the app crashed. Here is the log from Android Studio.
```
2019-11-27 20:38:02.873 14663-14707/org.tensorflow.lite.examples.detection E/AndroidRuntime: FATAL EXCEPTION: inference
    Process: org.tensorflow.lite.examples.detection, PID: 14663
    java.lang.IllegalArgumentException: Cannot convert between a TensorFlowLite buffer with 4915200 bytes and a Java Buffer with 1228800 bytes.
        at org.tensorflow.lite.Tensor.throwIfShapeIsIncompatible(Tensor.java:332)
        at org.tensorflow.lite.Tensor.throwIfDataIsIncompatible(Tensor.java:305)
        at org.tensorflow.lite.Tensor.setTo(Tensor.java:123)
        at org.tensorflow.lite.NativeInterpreterWrapper.run(NativeInterpreterWrapper.java:148)
        at org.tensorflow.lite.Interpreter.runForMultipleInputsOutputs(Interpreter.java:296)
        at org.tensorflow.lite.examples.detection.tflite.TFLiteObjectDetectionAPIModel.recognizeImage(TFLiteObjectDetectionAPIModel.java:197)
        at org.tensorflow.lite.examples.detection.DetectorActivity$2.run(DetectorActivity.java:181)
        at android.os.Handler.handleCallback(Handler.java:888)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:213)
        at android.os.HandlerThread.run(HandlerThread.java:67)
```
- Second: Observing the log above, I found that the problem may be because I modified the value to 640, so I changed a to 1280. The app still crashes, but a new error log appears. 
```
2019-11-27 20:58:10.236 17448-17553/org.tensorflow.lite.examples.detection E/AndroidRuntime: FATAL EXCEPTION: inference
    Process: org.tensorflow.lite.examples.detection, PID: 17448
    java.lang.ArrayIndexOutOfBoundsException: length=160; index=-6
        at java.util.Vector.elementData(Vector.java:734)
        at java.util.Vector.get(Vector.java:750)
        at org.tensorflow.lite.examples.detection.tflite.TFLiteObjectDetectionAPIModel.recognizeImage(TFLiteObjectDetectionAPIModel.java:218)
        at org.tensorflow.lite.examples.detection.DetectorActivity$2.run(DetectorActivity.java:181)
        at android.os.Handler.handleCallback(Handler.java:888)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:213)
        at android.os.HandlerThread.run(HandlerThread.java:67)
2019-11-27 20:58:10.243 17448-17448/org.tensorflow.lite.examples.detection E/ResMng NATIVE_MSG_FILTER: endActivityTransaction: margin state not match

```
The code in ```TFLiteObjectDetectionAPIModel.java:218``` is:
```
recognitions.add(
          new Recognition(
              "" + i,
              labels.get((int) outputClasses[0][i] + labelOffset),
              outputScores[0][i],
              detection));
```
Then I debug the App, I found the value of ```outputClasses[0][i]``` is negative, So maybe it is the reason . But I dont konw how to fix it. 
- Third: In this step, I modified the ```TF_OD_API_INPUT_SIZE``` parameter from 1280 to 640, and modified ```TF_OD_API_IS_QUANTIZED``` from ```true``` to ```false```. ```TF_OD_API_IS_QUANTIZED``` parameter is defined below ```TF_OD_API_INPUT_SIZE```. 
Unfortunately, the App still crashes with errors similar to those in Second.
```
2019-11-27 21:02:03.342 17894-18086/org.tensorflow.lite.examples.detection E/AndroidRuntime: FATAL EXCEPTION: inference
    Process: org.tensorflow.lite.examples.detection, PID: 17894
    java.lang.ArrayIndexOutOfBoundsException: length=160; index=-5
        at java.util.Vector.elementData(Vector.java:734)
        at java.util.Vector.get(Vector.java:750)
        at org.tensorflow.lite.examples.detection.tflite.TFLiteObjectDetectionAPIModel.recognizeImage(TFLiteObjectDetectionAPIModel.java:218)
        at org.tensorflow.lite.examples.detection.DetectorActivity$2.run(DetectorActivity.java:181)
        at android.os.Handler.handleCallback(Handler.java:888)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:213)
        at android.os.HandlerThread.run(HandlerThread.java:67)
2019-11-27 21:02:03.356 17894-17894/org.tensorflow.lite.examples.detection E/ResMng NATIVE_MSG_FILTER: endActivityTransaction: margin state not match
```
After the above three steps, I think it may be a problem with the android program, but I don't know how to modify the official demo to a code suitable for ```ssd_resnet_50_fpn_coco```.

**Comment by dao-kun** [comment]
did you solve it? I also met the same question

**Comment by srjoglekar246** [comment]
1. Has the model been trained on 640x640 images? You will find that in its `pipeline.config` file. If not, you should probably set the input size to those values.
2. I am assuming you changed the [model file](https://github.com/tensorflow/examples/blob/master/lite/examples/object_detection/android/app/src/main/java/org/tensorflow/lite/examples/detection/DetectorActivity.java#L55) path to `detect1.tflite`?
3. If the above two don't work, could you try using `ssd_mobilenet_v1_fpn_coco` from the model zoo which should give similar accuracies?

**Comment by yzmean** [comment]
> 1. Has the model been trained on 640x640 images? You will find that in its `pipeline.config` file. If not, you should probably set the input size to those values.
> 2. I am assuming you changed the [model file](https://github.com/tensorflow/examples/blob/master/lite/examples/object_detection/android/app/src/main/java/org/tensorflow/lite/examples/detection/DetectorActivity.java#L55) path to `detect1.tflite`?
> 3. If the above two don't work, could you try using `ssd_mobilenet_v1_fpn_coco` from the model zoo which should give similar accuracies?

1. The ```pipe;ine.config``` shows the model has been trained on 640x640. Hers is the ```pipeline.config```. Is there anything I misunderstood?
2. Sorry I forgot to mention model file path, yes I changed it to ```detect1.file```.
3. Thanks for your suggestion! :) I will try it ~

**Comment by yzmean** [comment]
> 1. Has the model been trained on 640x640 images? You will find that in its `pipeline.config` file. If not, you should probably set the input size to those values.
> 2. I am assuming you changed the [model file](https://github.com/tensorflow/examples/blob/master/lite/examples/object_detection/android/app/src/main/java/org/tensorflow/lite/examples/detection/DetectorActivity.java#L55) path to `detect1.tflite`?
> 3. If the above two don't work, could you try using `ssd_mobilenet_v1_fpn_coco` from the model zoo which should give similar accuracies?

- I have tried the `ssd_mobilenet_v1_fpn_coco` .and   change 300 to 640 . But the same error occurs. This time, I modified the code  `labels.get((int) outputClasses[0][i] + labelOffset),`([it is in here](https://github.com/tensorflow/examples/blob/e7aa7b80b09ad50884fdcf9ea1504a29a6123d7d/lite/examples/object_detection/android/app/src/main/java/org/tensorflow/lite/examples/detection/tflite/TFLiteObjectDetectionAPIModel.java#L213)) to ```labels.get((int) (outputClasses[0][i]<0?0:outputClasses[0][i]) + labelOffset),```.
Then it works !
- When I use the same way in `ssd_resnet_50_fpn_coco`, it still can't work,there is no boxes drawn in the screen :(

**Comment by srjoglekar246** [comment]
`ssd_resnet_50_fpn_coco` isn't that optimized for mobile inference, so I would suggest going with the mobilenet variant on this one. Is there any accuracy constraint that the mobilenet version doesn't satisfy for you?

**Comment by yzmean** [comment]
> `ssd_resnet_50_fpn_coco` isn't that optimized for mobile inference, so I would suggest going with the mobilenet variant on this one. Is there any accuracy constraint that the mobilenet version doesn't satisfy for you?

Thank you for your patience in answering my questions :) 

 `ssd_mobilenet_v1_fpn_coco` gives me satisfactory accuracy.However, my partner failed to retrain the model. Whether on the official dataset or our dataset, mAP is always 0. #7893

I would appreciate it if you could give me some help.

**Comment by srjoglekar246** [comment]
Looks like that issue is closed? Did the fix mentioned in the last comment on that issue work for you? If not, feel free to reopen that bug so it gets assigned to someone on the Object Detection team. I am closing this one. Thanks for the debugging efforts :-).

**Comment by yzmean** [comment]
> Looks like that issue is closed? Did the fix mentioned in the last comment on that issue work for you? If not, feel free to reopen that bug so it gets assigned to someone on the Object Detection team. I am closing this one. Thanks for the debugging efforts :-).


Unfortunately, that fix did not solve the problem. My companion has reopened the question, please help us, thank you.

**Comment by zychen2016** [comment]
Hello , how to retrain ssd_mobilenet_v2_fpn_coco, There is no pre-trained model in model_zoo

**Comment by eneshb** [comment]
Hey, did someone find a solution? I trained too a ssd fpn model but on Android it does not detecting.. The inference time is very high.
